zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,297 @@
+        return jsonify(1)
+    return jsonify(0)
+
+
+# creates a class in the data base. Requires form input (inv_code, class_name, class_language_id, max_students, teacher_id)
+@api.route("/add_class", methods=["POST"])
+@with_session
+def add_class():
+    inv_code = request.form.get("inv_code")
+    class_name = request.form.get("class_name")
+    class_language_id = request.form.get("class_language_id")
+    available_languages = Language.available_languages()
+    code_allowed = False
+    for code in available_languages:
+        if class_language_id in str(code):
+            code_allowed = True
+
+    if not code_allowed:
+        flask.abort(400)
+    class_language = Language.find_or_create(class_language_id)
+    teacher_id = flask.g.user.id
+    max_students = request.form.get("max_students")
+    if int(max_students) < 1:
+        flask.abort(400)
+
+    try:
+        c = Cohort(inv_code, class_name, class_language, max_students)
+        db.session.add(c)
+        db.session.commit()
+        _link_teacher_class(teacher_id, c.id)
+        return jsonify(1)
+    except ValueError:
+        # print("value error")
+        flask.abort(400)
+    except sqlalchemy.exc.IntegrityError:
+        # print("integ error")
+        flask.abort(400)
+
+
+# creates user and adds them to a cohort
+@api.route("/add_user_with_class", methods=['POST'])
+def add_user_with_class():
+    email = request.form.get("email")
+    password = request.form.get("password")
+    username = request.form.get("username")
+    inv_code = request.form.get("inv_code")
+
+    try:
+        cohort_id = Cohort.get_id(inv_code)
+        cohort = Cohort.find(cohort_id)
+    except:
+        flask.abort(400)
+
+    if not len(password) == 0:
+        # Checks to see if there is space, if there is it increments the cur_students variables.
+        # Therefore it is essential that you ensure you add the student if this function returns true.
+        # Hence it being placed after other checks.
+        # However, if an exeption is caught, this error is handled.
+        if cohort.request_join():
+            try:
+                user = User(email, username, password)
+                user.cohort = cohort
+                db.session.add(user)
+                db.session.commit()
+                return 'user created'
+            except ValueError:
+                cohort.undo_join()
+                flask.abort(400)
+            except sqlalchemy.exc.IntegrityError:
+                cohort.undo_join()
+                flask.abort(400)
+        return 'no more space in class!'
+    flask.abort(400)
+
+
+# Get user bookmarks
+@api.route("/get_user_bookmarks/<id>", methods=["GET", "POST"])
+@with_session
+def get_user_bookmarks(id):
+    user = User.query.filter_by(id=id).one()
+    if (not _has_permission_for_class(user.cohort_id)):
+        flask.abort(401)
+    # True input causes function to return context too.
+    return json_result(user.bookmarks_by_day(True))
```

Comment:
i alway forget what's the difference between this one
and bookmarks by day. could we name it `cohort_member_bookmarks`? 
and add a comment that mentions that this is different than the other
one because it's used from the perspective of the teacher.

finally, why is POST a possible method here? it clearly should only be GET.